process/block/preprocess/rewardsHandler.go

```diff
@@ -21,18 +22,20 @@ const burnPercentage = 0.5      // 1 = 100%, 0 = 0%
 var rewardValue = big.NewInt(1000)
 
 type rewardsHandler struct {
-    address         process.SpecialAddressHandler
-    hasher          hashing.Hasher
-    marshalizer     marshal.Marshalizer
-    mut             sync.Mutex
-    accumulatedFees *big.Int
+    address          process.SpecialAddressHandler
+    shardCoordinator sharding.Coordinator
+    hasher           hashing.Hasher
+    marshalizer      marshal.Marshalizer
+    mut              sync.Mutex
+    accumulatedFees  *big.Int
 
     rewardTxsFromBlock map[string]*rewardTx.RewardTx
 }
 
 // NewRewardTxHandler constructor for the reward transaction handler
 func NewRewardTxHandler(
```

Comment:
Add IsInterfaceNil() checks for every parameter. And also check if shardCoordinator == nil || shardCoordinator.IsInterfaceNil()

Reply:
done